cmd/otelcol/main.go

```diff
@@ -169,66 +155,94 @@ func checkRuntimeParams() {
+// file path flag, config file path environment variable and config YAML environment variable are not specified.
+func setConfigPath() {
+	configPathFlag := getKeyValue(os.Args[1:], "--config")
+	configPathVar := os.Getenv(configEnvVarName)
+	configYaml := os.Getenv(configYamlEnvVarName)
+
+	if configPathFlag == "" && configPathVar == "" {
+		if configYaml == "" {
+			setConfigPathToDefault()
+		} else {
+			log.Printf("Configuring using %s environment variable YAML", configYamlEnvVarName)
 		}
-		_, err = os.Stat(defaultLocalSAPMConfig)
-		if err == nil {
-			config = defaultLocalSAPMConfig
+		return
+	}
+
+	if configPathFlag != "" {
+		if configPathVar != "" && configPathVar != configPathFlag {
+			log.Printf("Both %v and '--config' were specified. Overriding %q environment variable value with %q for this session", configEnvVarName, configPathVar, configPathFlag)
 		}
-		if config == "" {
-			log.Fatalf("Unable to find the default configuration file, ensure %s environment variable is set properly", configEnvVarName)
+		os.Setenv(configEnvVarName, configPathFlag)
+
+		if configYaml != "" {
+			log.Printf("Both %s and '--config' were specified. Using '--config' value %s for this session", configYamlEnvVarName, configPathFlag)
 		}
-	} else {
-		// Check if file exists.
-		_, err := os.Stat(config)
-		if err != nil {
-			log.Fatalf("Unable to find the configuration file (%s) ensure %s environment variable is set properly", config, configEnvVarName)
+
+		log.Printf("Set config to %v", configPathFlag)
+		return
+	}
+
+	if configPathVar != "" {
+		if !contains(os.Args[1:], "--config") {
+			// Inject the command line flag that controls the configuration.
+			os.Args = append(os.Args, "--config="+configPathVar)
+		}
+
+		if configYaml != "" {
+			log.Printf("Both %s and %s were specified. Using %s environment variable value %s for this session", configEnvVarName, configYamlEnvVarName, configEnvVarName, configPathVar)
 		}
+
+		log.Printf("Set config to %v", configPathVar)
 	}
+}
 
-	switch config {
-	case
-		defaultDockerSAPMConfig,
-		defaultDockerOTLPConfig,
-		defaultLocalSAPMConfig,
-		defaultLocalOTLPConfig:
-		// The following environment variables are required.
-		// If any are missing stop here.
-		requiredEnvVars := []string{realmEnvVarName, tokenEnvVarName}
-		for _, v := range requiredEnvVars {
-			if len(os.Getenv(v)) == 0 {
-				log.Printf("Usage: %s=12345 %s=us0 %s", tokenEnvVarName, realmEnvVarName, os.Args[0])
-				log.Fatalf("ERROR: Missing required environment variable %s with default config path %s", v, config)
-			}
+func setConfigPathToDefault() {
+	// Checking environment variables required by default config files.
+	for _, v := range []string{realmEnvVarName, tokenEnvVarName} {
+		if len(os.Getenv(v)) == 0 {
+			log.Printf("Usage: %s=12345 %s=us0 %s", tokenEnvVarName, realmEnvVarName, os.Args[0])
+			log.Fatalf("ERROR: Missing required environment variable %s", v)
```

Comment:
It was intentional that only default config paths are applicable for this requirement, even if specifically set by the user.  If this is to change I'd do it in another PR.

Reply:
Yes, checking for this requirement only applies when setting default config. This chunk is in setConfigPathToDefault(). Maybe I should move the checking to the end of the function